go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -98,7 +102,17 @@ type DefaultNetworkController struct {
 	// Controller used to handle the admin policy based external route resources
 	apbExternalRouteController *apbroutecontroller.ExternalGatewayMasterController
 
-	egressFirewallDNS *EgressDNS
+	// Controller used to handle dns name resolver resoures
+	dnsNameResolverController controller.Controller
+	// Lister for dns name resolver
+	dnsLister ocpnetworklisterv1alpha1.DNSNameResolverLister
+	// maps to hold the details of the DNS names and the corresponding
+	// DNSNameResolver object names
+	dnsNameObj map[string]string
+	objDNSName map[string]string
+
+	egressFirewallDNS         *dnsnameresolver.EgressDNS
+	egressFirewallExternalDNS *dnsnameresolver.ExternalEgressDNS
```

Comment:
and also move all internals to the resolver implementation, so that DefaultNetworkController has only 1 field for dnsResolver (no matter which implementation it's gonna be?)

Reply:
I have created an interface which both EgressDNS and ExternalEgressDNS implements. The interface is only added to the DefaultNetworkController.